test/ocsp/helper/helper.go

```diff
@@ -0,0 +1,187 @@
+			return nil, nil
+		} else {
+			return nil, fmt.Errorf("certificate expired %s ago: %s",
+				time.Now().Sub(cert.NotAfter), cert.NotAfter)
+		}
+	}
+	issuer, err := getIssuer(cert)
+	if err != nil {
+		return nil, fmt.Errorf("getting issuer: %s", err)
+	}
+	req, err := ocsp.CreateRequest(cert, issuer, nil)
+	if err != nil {
+		return nil, fmt.Errorf("creating OCSP request: %s", err)
+	}
+	if len(cert.OCSPServer) == 0 {
+		return nil, fmt.Errorf("no ocsp servers in cert")
+	}
+	encodedReq := base64.StdEncoding.EncodeToString(req)
+	var httpResp *http.Response
+	ocspServer := cert.OCSPServer[0]
+	ocspURL, err := url.Parse(ocspServer)
+	if *urlOverride != "" {
+		ocspServer = *urlOverride
+	}
+	if err != nil {
+		return nil, fmt.Errorf("parsing URL: %s", err)
+	}
+	http.DefaultClient.Timeout = 5 * time.Second
+	if *method == "GET" {
+		ocspURL.Path = encodedReq
+		fmt.Printf("Fetching %s\n", ocspURL.String())
+		var err error
+		httpResp, err = http.Get(ocspURL.String())
+		if err != nil {
+			return nil, fmt.Errorf("fetching: %s", err)
+		}
+	} else if *method == "POST" {
+		fmt.Printf("POSTing request, reproduce with: curl -i --data-binary @- %s < <(base64 -d <<<%s)\n",
+			ocspServer, encodedReq)
+		var err error
+		httpResp, err = http.Post(ocspServer, "application/ocsp-request", bytes.NewBuffer(req))
+		if err != nil {
+			return nil, fmt.Errorf("fetching: %s", err)
+		}
+	} else {
+		return nil, fmt.Errorf("invalid method %s, expected GET or POST", *method)
+	}
+	fmt.Printf("HTTP %d\n", httpResp.StatusCode)
+	for k, v := range httpResp.Header {
+		for _, vv := range v {
+			fmt.Printf("%s: %s\n", k, vv)
+		}
+	}
+	if httpResp.StatusCode != 200 {
+		return nil, fmt.Errorf("http status code %d", httpResp.StatusCode)
+	}
+	respBytes, err := ioutil.ReadAll(httpResp.Body)
+	if err != nil {
+		return nil, err
+	}
+	if len(respBytes) == 0 {
+		return nil, fmt.Errorf("empty reponse body")
+	}
+	fmt.Printf("\nDecoding body: %s\n", base64.StdEncoding.EncodeToString(respBytes))
+	resp, err := ocsp.ParseResponseForCert(respBytes, cert, issuer)
+	if err != nil {
+		return nil, fmt.Errorf("parsing response: %s", err)
+	}
+	if resp.Status != *expectStatus {
+		return nil, fmt.Errorf("wrong CertStatus %d, expected %d", resp.Status, *expectStatus)
+	}
+	fmt.Printf("\n")
+	fmt.Printf("Good response:\n")
+	fmt.Printf("  CertStatus %d\n", resp.Status)
+	fmt.Printf("  SerialNumber %036x\n", resp.SerialNumber)
+	fmt.Printf("  ProducedAt %s\n", resp.ProducedAt)
+	fmt.Printf("  ThisUpdate %s\n", resp.ThisUpdate)
+	fmt.Printf("  NextUpdate %s\n", resp.NextUpdate)
+	fmt.Printf("  RevokedAt %s\n", resp.RevokedAt)
+	fmt.Printf("  RevocationReason %d\n", resp.RevocationReason)
+	fmt.Printf("  SignatureAlgorithm %s\n", resp.SignatureAlgorithm)
+	fmt.Printf("  Extensions %#v\n", resp.Extensions)
+	timeTilExpiry := resp.NextUpdate.Sub(time.Now())
+	if timeTilExpiry < tooSoonDuration {
+		return nil, fmt.Errorf("NextUpdate is too soon: %s", timeTilExpiry)
```

Comment:
nit: Should this happen before we print out all of the stuff above? Given that you print out this is a `Good response` it seems slightly confusing.